hpx/runtime/threads/detail/scheduled_thread_pool_impl.hpp

```diff
@@ -1460,13 +1533,19 @@ namespace hpx { namespace threads { namespace detail
             oldstate == state_stopped);
 
         resource::get_partitioner().unassign_pu(id_.name(), virt_core);
+        compat::thread t;
         std::swap(threads_[virt_core], t);
 
         if (threads::get_self_ptr())
         {
-            while (virt_core == hpx::get_worker_thread_num())
+            std::size_t thread_num = thread_offset_ + virt_core;
+
+            for (std::size_t k = 0;
+                 thread_num == hpx::get_worker_thread_num();
+                 ++k)
             {
-                hpx::this_thread::suspend();
+                util::detail::yield_k(k,
+                    "scheduled_thread_pool::remove_processing_unit_internal");
             }
         }
```

Comment:
This loop looks like to be a good candidate for being extracted into a utility function (similar construct are used during shutdown as well, iirc).

Reply:
You're right. The loops at shutdown are at least similar, I'll see what common functionality I can get out of there.

Reply:
I've pulled out a `yield_while` function into `yield_k.hpp`, which takes a predicate for terminating the loop and some optional arguments (same as in `yield_k`). It doesn't really make the code shorter but I think the intention is clearer, and `k` is now hidden which I think is good.